packages/nimble-components/src/table/index.ts

```diff
@@ -336,35 +373,103 @@ export class Table<
             return { ...record };
         });
         this.validateWithData(data);
+        this.updateTableOptions(
+            {
+                data,
+                state: {
+                    expanded: true
```

Comment:
waiting to review the state management in the top-level table pending the discussions mentioned. Looked through other bits though.

Reply:
Worked with @mollykreis to converge on the idea of us providing an override implementation for TanStack's `getIsExpanded()` method on a `Row`. This, coupled with us maintaining some extra state for keeping tracks of which rows have been explicitly collapsed, and also setting the default for `expanded` in the TanStack state to `true`, allowed us to avoid updating the TanStack state twice (with the first one set to not refresh the rows).

The concessions with the current implementation are:
- Setting `expanded` to true by default, and when the groups change.
    - This is due to how [TanStack currently implements how to get the set of current expanded row models](https://github.com/TanStack/table/blob/134414e7c5c6ff732c839a6c8b8f43a933967d3e/packages/table-core/src/utils/getExpandedRowModel.ts#L17). It seems possible, and likely, that this implementation should change to simply have it call `getIsExpanded()` on the available `rowModel`. I plan on filing an issue with TanStack on this.
- We currently can have stale state kept around in the `collapsedRows` set. Namely when the data changes, and any prior collapsed row ids are no longer in the current set of data.
    - We could definitely do the work to clear out this state when the data changes, but that seems, in the worst case scenario fairly expensive (recognizing that not cleaning it up is effectively a memory leak). I do intend to implement the solution that would clear the state, and profile it, but wanted to move forward with this discussion before spending time on that.